tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -186,11 +196,22 @@ spec:
             build_sha=$(echo "$IMAGE_REF" | cut -d ':' -f 2)
             passedTimestampFormat=$(jq -r --arg default "$defaultTimestampFormat" \
               '.timestampFormat // $default' <<< "$component")
-            timestamp="$(date -d "$currentTimestamp" "+$passedTimestampFormat")"
+            release_timestamp="$(date -d "$currentTimestamp" "+$passedTimestampFormat")"
+            arch_json="$(get-image-architectures "${IMAGE_REF}")"
+            # The build-date label is not the same per architecture, but we don't support separate tags per arch. So,
+            # we just use inspect the first digest listed
+            arch="$(jq -rs 'map(.platform.architecture) | .[0]' <<< "$arch_json")"
+            os="$(jq -rs 'map(.platform.os) | .[0]' <<< "$arch_json")"
+            build_date="$(skopeo inspect --no-tags --override-os "${os}" --override-arch "${arch}" \
+              docker://"${IMAGE_REF}" | jq -r '.Labels."build-date"')"
+            timestamp="$(date -d "${build_date}" "+$passedTimestampFormat")"
```

Comment:
Nice! I thought it might be hard to parse the date string, but it seems `-d` is able to parse anything as long as it's a reasonable date string. Cool.